src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,159 @@
+                    diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureServices.Id:
+                    context.RegisterCodeFix(
+                    CodeAction.Create("Fix references to Services properties on WebApplicationBuilder",
+                    cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, "Services"),
+                    equivalenceKey: DiagnosticDescriptors.DoNotUseHostConfigureServices.Id),
+                    diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id:
+                    context.RegisterCodeFix(
+                    CodeAction.Create("Fix references to Services properties on WebApplicationBuilder",
+                    cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, "Configuration"),
+                    equivalenceKey: DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id),
+                    diagnostic);
+                    break;
+            }
+        }
+
+        return Task.CompletedTask;
+    }
+
+    private static async Task<Document> FixWebApplicationBuilder(Diagnostic diagnostic, Document document, CancellationToken cancellationToken, string IDENTIFIER_METHOD_NAME)
+    {
+        DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken);
+        var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root == null)
+        {
+            return document;
+        }
+
+        var diagnosticTarget = root.FindNode(diagnostic.Location.SourceSpan); //whole invocation starting from "builder"
+
+        if (diagnosticTarget is InvocationExpressionSyntax { } originalInvocation)
+        {
+            if (originalInvocation.Expression is not MemberAccessExpressionSyntax hostBasedInvocationMethodExpr
+                || hostBasedInvocationMethodExpr.Expression is not MemberAccessExpressionSyntax subExpression)
+            {
+                return editor.OriginalDocument;
+            }
+
+            var identifierMethod = SyntaxFactory.IdentifierName(IDENTIFIER_METHOD_NAME);
+
+            subExpression = subExpression.WithName(identifierMethod);
+            var indentation = hostBasedInvocationMethodExpr.GetLeadingTrivia();
+            hostBasedInvocationMethodExpr = hostBasedInvocationMethodExpr.WithExpression(subExpression)
+                .NormalizeWhitespace().WithLeadingTrivia(indentation); // replace Host/WebHost with identifierMethod
+
+            if (originalInvocation.ArgumentList.Arguments.Count == 0)
+            {
+                return editor.OriginalDocument;
+            }
+
+            var initArgumentExpr = originalInvocation.ArgumentList.Arguments[0].Expression; // builder => { }
```

Comment:
Could you double-check that this works as expected when named parameters are used, and arguments are out of order? I think it might break.

Reply:
Yep, I learned from https://github.com/dotnet/aspnetcore/pull/42757 that you have to check parameter optionality when you use argument list.

Good to call out named parameters here. I'm not sure how that would affect the behavior. It would be good to add a test for this.